### ISSUE TITLE
New rule: UseSumInsteadOfFlatMapSize

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5399]
## Context / Expected Behavior of the rule
Similar to `UseAnyOrNoneInsteadOfFind`, it would recognize cases when we are doing `flatMap` instead of calculating the end result directly.

```kotlin
val size = lambdaExpression.collectDescendantsOfType<KtBlockExpression>().flatMap { it.statements }.size
```
[could be](https://github.com/detekt/detekt/pull/5397/files#r991027678)
```kotlin
val size = lambdaExpression.collectDescendantsOfType<KtBlockExpression>().sumOf { it.statements.size }
```
which is simpler and more performant.

I've seen this flatMap + size/isEmpty pattern used a lot previously, even by myself, but we can do better!

So far I can only think of
 * `.flatMap{}.size` -> `.sumOf`
 * `.flatMap{}.isEmpty()` -> could be `.sumOf`, but can be even shortcut with `.none{it.isNotEmpty()}`
 * `.flatMap{}.isNotEmpty()` -> could be `.sumOf`, but can be shortcut with `.any{it.isNotEmpty()}`

There might be other cases too, I think the first one would be good enough to flag, as it has a clearly better replacement. The last 2 might be too complex for the potentially micro-optimization.

**Comment by BraisGabin** [comment]
This rule shouldn't be that difficult to implement. It should be just a variation of `UseAnyOrNoneInsteadOfFind`
